### ISSUE TITLE
Gradle not hiding captured output when running in parallel

### ISSUE THREAD

**Issue by jaredsburrows** [gradle/gradle#2466]
Based on this issue: https://discuss.gradle.org/t/logging-capturestandardoutput-does-not-capture-all-stdout-when-running-in-parallel/1147 which leads to this issue: https://issues.gradle.org/browse/GRADLE-3000, I decided to file this issue on Gradle's github. I am not sure if this plugin is using the correct API to capture standard output or there is a bug in Gradle for parallel builds.

### Expected Behavior
Gradle task should capture log output when running in parallel. See this Github issue: https://github.com/apollographql/apollo-android/issues/549 and the PR to try and fix this issue: https://github.com/apollographql/apollo-android/pull/323. The "fix" is using `getLogging().captureStandardOutput(LogLevel.INFO);`. Should the plugin be using https://github.com/gradle/gradle/blob/master/design-docs/features/composite-build/tooling-api/capture-output.md#overview?

### Current Behavior
Gradle task log output is shown when running in parallel.

### Context
When the output from the task is shown, they show up as errors in the IDE.

### Steps to Reproduce (for bugs)
Set up plugin and run your build with `--parallel`

**Comment by bmuschko** [comment]
Maybe we can start by reproducing the issue with a simple example? The issue seems to surface in a relatively complex environment: A plugin applying another plugin, that plugin calls of to Node and then the error showing up in Android Studio. It's somewhat hard to follow what's going on here.

> The "fix" is using `getLogging().captureStandardOutput(LogLevel.INFO);`.

Is this working? I believe what happens is that any standard output will only be rendered when you run the build with `INFO` log level. I believe you'll just run into https://issues.gradle.org/browse/GRADLE-3000 which needs fixing.

> Should the plugin be using https://github.com/gradle/gradle/blob/master/design-docs/features/composite-build/tooling-api/capture-output.md#overview?

Sorry, I am not familiar with that spec or the functionality but I don't think you'll want to use the tooling API for that purpose.

**Comment by jaredsburrows** [comment]
@bmuschko You are right about the complex example. `getLogging().captureStandardOutput(LogLevel.INFO);` seems to only work if I turn off Gradle's parallel compilation. 

I also believe you are correct about https://issues.gradle.org/browse/GRADLE-3000. I am not sure if the new spec is implemented or not. Should I make a new issue for Gradle-3000 or is there already one on Github for it?

**Comment by bmuschko** [comment]
I'd suggest we try to see if the underlying issue is the one described in `GRADLE-3000` and then change this issue accordingly.

**Comment by jaredsburrows** [comment]
@bmuschko I believe it is that issue. However, after upgrading to Gradle `4.1-rc-1`, here is all that I am seeing:

```
> Task :yammer-data:installApolloCodegen
apollo-android@0.0.1 /Users/jaburrow/repo/android2/yammer-data/build/apollo-codegen
`-- apollo-codegen@0.12.10 
```
https://github.com/apollographql/apollo-android/issues/549#issuecomment-317815580

**Comment by cdancy** [comment]
Any updates here? We're seeing the same on cloudbees-jenkins builds with a lot of logging getting suppressed.
